packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol

```diff
@@ -0,0 +1,326 @@
+    function test_withdraw_fromCustomGasTokenChain_fails() public {
+        // Arrange
+        uint256 amount = 1000;
+        vm.deal(alice, amount);
+        vm.prank(alice);
+        superchainWeth.deposit{ value: amount }();
+        vm.mockCall(address(l1Block), abi.encodeCall(l1Block.isCustomGasToken, ()), abi.encode(true));
+
+        // Act
+        vm.prank(alice);
+        vm.expectRevert(NotCustomGasToken.selector);
+        superchainWeth.withdraw(amount);
+
+        // Assert
+        assertEq(alice.balance, 0);
+        assertEq(superchainWeth.balanceOf(alice), amount);
+    }
+
+    /// @notice Tests that the sendERC20 function can be called with a sufficient balance on a
+    ///         non-custom gas token chain. Also tests that the proper calls are made, ETH is
+    ///         burned, and the proper events are emitted.
+    function test_sendERC20_sufficientBalance_succeeds() public {
+        // Arrange
+        uint256 amount = 1000;
+        uint256 chainId = 1;
+        vm.deal(alice, amount);
+        vm.prank(alice);
+        superchainWeth.deposit{ value: amount }();
+
+        // Act
+        vm.expectEmit(address(superchainWeth));
+        emit Transfer(alice, address(0), amount);
+        vm.expectEmit(address(superchainWeth));
+        emit SendERC20(alice, bob, amount, chainId);
+        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(ETHLiquidity.burn, ()), 1);
+        vm.expectCall(
+            Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER,
+            abi.encodeCall(
+                IL2ToL2CrossDomainMessenger.sendMessage,
+                (chainId, address(superchainWeth), abi.encodeCall(superchainWeth.relayERC20, (bob, amount)))
+            ),
+            1
+        );
+        vm.prank(alice);
+        superchainWeth.sendERC20(bob, amount, chainId);
+
+        // Assert
+        assertEq(alice.balance, 0);
+        assertEq(superchainWeth.balanceOf(alice), 0);
+    }
+
+    /// @notice Tests that the sendERC20 function can be called with a sufficient balance on a
+    ///         custom gas token chain. Also tests that the proper calls are made and the proper
+    ///         events are emitted but ETH is not burned via the ETHLiquidity contract.
+    function test_sendERC20_sufficientFromCustomGasTokenChain_succeeds() public {
+        // Arrange
+        uint256 amount = 1000;
+        uint256 chainId = 1;
+        vm.deal(alice, amount);
+        vm.prank(alice);
+        superchainWeth.deposit{ value: amount }();
+        vm.mockCall(address(l1Block), abi.encodeCall(l1Block.isCustomGasToken, ()), abi.encode(true));
+
+        // Act
+        vm.expectEmit(address(superchainWeth));
+        emit Transfer(alice, address(0), amount);
+        vm.expectEmit(address(superchainWeth));
+        emit SendERC20(alice, bob, amount, chainId);
+        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(ETHLiquidity.burn, ()), 0);
```

Comment:
Should this burn be happening on a custom gas token chain? Intuitively I don't think that cgt chains should interact at all with ETHLiquidity

Reply:
Ah here the `expectCall` is asserting that the call does *not* happen for custom gas token (last parameter is zero).